packages/modules/viewer-react/src/components/app-ui/providers/EcsqlWidgetProvider.tsx

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import type {
+  AbstractWidgetProps,
+  UiItemsProvider,
+} from "@itwin/appui-abstract";
+import { StagePanelLocation, StagePanelSection } from "@itwin/appui-abstract";
+import React from "react";
+
+import EcsqlWidget from "../EcsqlWidget";
+
+export class EcsqlWidgetProvider implements UiItemsProvider {
+  public readonly id = "EcsqlWidgetProvider";
+
+  public provideWidgets(
+    stageId: string,
```

Comment:
Take a look at what we do for stageId and stageUsage in the other providers in viewer-components-react